MovieApp/MovieApp/App/Features/Movies/DataSources/RealmDataSource/RealmDataSourceProtocol.swift

```diff
@@ -0,0 +1,5 @@
+protocol RealmDataSourceProtocol {
+
+    func saveData(model: [MovieDataSourceModel], category: RealmCategory)
```

Comment:
`RealmDataSource` should not be aware of other Data Source models, only of `RealmDataSourceModel`.